src/validation.h

```diff
@@ -557,7 +557,14 @@ static const unsigned int REJECT_CONFLICT = 0x102;
 /** Dump the mempool to disk. */
 void DumpMempool();
 
+/** Result states for loading the mempool from disk */
+enum LoadMempoolResult {
```

Comment:
I think this is a bit overkill. `bool` is there already. And returning `true` only in case of successful `mempool.dat` load is enough. The only question is what to return in case of corrupted `mempool.dat`. Returning `true` instead of `false` makes sense here. And when `true` is returned, dump mempool at shutdown.